src/org/opendatakit/briefcase/ui/MainBriefcaseWindow.java

```diff
@@ -213,9 +214,18 @@ public void setFullUIEnabled(boolean enabled) {
      * Create the application.
      */
     private MainBriefcaseWindow() {
-        briefcaseAnalytics.trackStartup();
-
         frame = new JFrame();
+
+        Analytics analytics = Analytics.from(
+            BuildConfig.GOOGLE_TRACKING_ID,
+            BuildConfig.VERSION,
+            BriefcasePreferences.getUniqueUserID(),
+            Toolkit.getDefaultToolkit().getScreenSize(),
+            frame::getSize
+        );
+        analytics.enter("Briefcase");
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> analytics.leave("Briefcase")));
```

Comment:
How does the thread help us here?

Reply:
I can't call `Runtime.addShutdownHook(Thread hook)` otherwise... I agree that it would be much nicer to just pass a `Runnable`.

Any alternatives?

Reply:
Sorry, I missed that addShutdownHook takes a Thread.

BTW, I played around with using listeners instead of this hackier approach, but it seems you have to do a bunch of Mac-specific stuff to hook into Quit (as opposed to clicking the close button in the JFrame).